amplify/backend/function/postUser/src/index.py

```diff
@@ -10,27 +10,27 @@
 db = client.Client(f"wss://{endpoint}:{port}/gremlin", "g")
 
 def lambda_handler(event, context):
-    userData = event["userInfo"]
-    firstName = userData["firstName"]
-    lastName = userData["lastName"]
-    dob = userData["DOB"]
-    username = userData["username"]
+    userData = event["request"]["userAttributes"]
+    # Inputs currently required by cognito
+    firstName = userData["given_name"]
+    lastName = userData["family_name"]
+    username = userData["preferred_username"]
     email = userData["email"]
-    gender = userData["gender"]
-    bio = userData["bio"]
 
+    # Inputs not required by cognito
+    dob = ""
+    gender = ""
+    bio = ""
+
+    # Add a user
     query = f"g.addV('user').property('firstName', '{firstName}').property('lastName', '{lastName}').property('DOB', '{dob}').property('username', '{username}').property('email', '{email}').property('gender', '{gender}').property('bio', '{bio}');"
```

Comment:
Remove variables since they are just empty strings and reorder the query.
```suggestion
    # Creates a user vertex using attributes provided by Cognito and initializing the
    # remaining attributes to empty strings (users can update these attributes later)
    query = f"g.addV('user').property('firstName', '{firstName}').property('lastName', '{lastName}').property('username', '{username}').property('email', '{email}').property('DOB', '').property('gender', '').property('bio', '');"
```

Reply:
We need to leave them blank, the way I have the rest of the API set up, we can't add attributes as we please it all needs to be done in postUser. This way users can't freely add attributes to their users, they can only edit existing ones.